app/services/recommendations_service/media.rb

```diff
@@ -16,6 +16,66 @@ def poll
     def recommendations_for(klass)
       res = client.get("recommendations/#{user.id}/")
       reccs = res.body.dig('results', 'recommendations', 'media', klass.name)
+      enrich(klass, reccs)
+    end
+
+    def category_recommendations_for(klass)
+      res = client.get("recommendations/#{user.id}/")
+      reccs = res.body.dig(
+        'results',
+        'recommendations',
+        'media',
+        'Genre Recommendations'
+      )
+      enrich_category(klass, reccs)
+    end
+
+    def realtime_recommendations_for(klass)
+      res = client.get("realtime/#{user.id}/")
+      reccs = res.body.dig('results', klass.name)
+      enrich(klass, reccs)
+    end
+
+    def realtime_category_recommendations_for(klass)
+      res = client.get("realtime/#{user.id}/")
+      reccs = res.body.dig('results', 'Genre Recommendations')
+      enrich_category(klass, reccs)
+    end
+
+    def strength
+      res = client.get("profile_strength/#{user.id}/")
+      strengths = res.body.dig('results', 'profile_strength')
+      strengths
+    end
+
+    def enrich_category(klass, reccs)
+      reccs = reccs.reduce(&:merge)
+      field_name = klass.model_name.i18n_key.to_s
+      field_key = field_name.to_sym
+
+      categories_objects = []
+      reccs.each do |key, val|
+        val = val.reduce(&:merge)
+        media_ids = val.keys.map do |v_key|
+          v_key.split(':').last.to_i if v_key.include? klass.model_name
+        end
+        media_ids.compact
+        category_object = Category.includes(
+          field_key
+        ).where(
+          field_key => { id: media_ids }
+        ).find_by(id: key.to_i)
+        next unless category_object
+        j_category_object = category_object.as_json(include: field_key)
+        j_category_object[field_name].sort_by do |e|
+          media_ids.index(e[:id]) || media_ids.length
+        end
+        categories_objects << j_category_object
+      end
+      categories_objects
+    end
```

Comment:
The input: https://kitsu.slack.com/files/kev/F5R490W8N/-.py
The output is the json serialization of  of a category openstructs, with the anime sorted